dags/hivemind_etl_helpers/gdrive_ingestion_etl.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+from hivemind_etl_helpers.src.utils.credentials import load_redis_credentials
+from llama_index.core.ingestion import (
+    DocstoreStrategy,
+    IngestionCache,
+    IngestionPipeline,
+)
+from llama_index.core.node_parser import SemanticSplitterNodeParser
+from llama_index.storage.docstore.postgres import PostgresDocumentStore
+from llama_index.storage.kvstore.redis import RedisKVStore as RedisCache
+from llama_index.vector_stores.postgres import PGVectorStore
+from tc_hivemind_backend.db.credentials import load_postgres_credentials
+from tc_hivemind_backend.embeddings.cohere import CohereEmbedding
+
+
+class GoogleDriveIngestionPipeline:
+    def __init__(self, community_id: str):
+        self.community_id = community_id
+
+        # Embedding models
+        self.cohere_model = CohereEmbedding()
+        self.pg_creds = load_postgres_credentials()
+        self.redis_cred = load_redis_credentials()
+        self.table_name = "gdrive"
+        self.dbname = f"community_{community_id}"
+
+        # Database details
+        self.redis_host = self.redis_cred["host"]
+        self.redis_port = self.redis_cred["port"]
+
+    def run_pipeline(self, docs):
+        pipeline = IngestionPipeline(
+            transformations=[
+                SemanticSplitterNodeParser(embed_model=self.cohere_model),
+                self.cohere_model,
+            ],
+            docstore=PostgresDocumentStore.from_params(
+                host=self.pg_creds["host"],
+                port=self.pg_creds["port"],
+                database=self.dbname,
+                user=self.pg_creds["user"],
+                password=self.pg_creds["password"],
+                table_name=self.table_name + "_docstore",
+            ),
+            vector_store=PGVectorStore.from_params(
+                host=self.pg_creds["host"],
+                port=self.pg_creds["port"],
+                database=self.dbname,
+                user=self.pg_creds["user"],
+                password=self.pg_creds["password"],
+                table_name=self.table_name,
+                embed_dim=1024,
```

Comment:
Please don't hardcode the `embed_dim`. You can load it via 
```
from tc_hivemind_backend.db.utils.model_hyperparams import load_model_hyperparams
```

see usage example

https://github.com/TogetherCrew/airflow-dags/blob/main/dags/hivemind_etl_helpers/discourse_summary_etl.py#L90